src/App.tsx

```diff
@@ -1,61 +1,133 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { ToDo } from './components/type.ts/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId)
+    || undefined;
+}
+
+export const initialTodos: ToDo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const getToDoId = () => {
+    return todos.reduce((id, todo) => Math.max(id, todo.id), 0) + 1;
+  };
+
+  const handleOnSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setTitleError(!title);
```

Comment:
I can add an empty todo using spaces. I suppose you can fix it with a couple of chars of code here and below🙃